x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/create_security_rule_type_wrapper.ts

```diff
@@ -492,30 +485,40 @@ export const createSecurityRuleTypeWrapper: CreateSecurityRuleTypeWrapper =
                 alertsCreated: createdSignalsCount > 0,
                 disabledActions,
               });
-              if (result.warningMessages.length) {
-                result.warningMessages.push(disabledActionsWarning);
-              } else {
-                warningMessage = [
-                  ...(warningMessage ? [warningMessage] : []),
-                  disabledActionsWarning,
-                ].join(', ');
-                wroteWarningStatus = true;
-              }
+              wrapperWarnings.push(disabledActionsWarning);
             }
 
-            if (result.warningMessages.length) {
+            if (result.warningMessages.length > 0 || wrapperWarnings.length > 0) {
+              // write warning messages first because if we have still have an error to write
+              // we want to write the error messages last, so that the errors are set
+              // as the current status of the rule.
               await ruleExecutionLogger.logStatusChange({
                 newStatus: RuleExecutionStatusEnum['partial failure'],
-                message: truncateList(result.warningMessages).join(', '),
+                message: truncateList(result.warningMessages.concat(wrapperWarnings)).join(', '),
                 metrics: {
                   searchDurations: result.searchAfterTimes,
                   indexingDurations: result.bulkCreateTimes,
                   enrichmentDurations: result.enrichmentTimes,
                 },
               });
             }
-
-            if (result.success) {
+            if (wrapperErrors.length > 0 || result.errors.length > 0) {
+              await ruleExecutionLogger.logStatusChange({
+                newStatus: RuleExecutionStatusEnum.failed,
+                message: truncateList(result.errors.concat(wrapperErrors)).join(', '),
+                metrics: {
+                  searchDurations: result.searchAfterTimes,
+                  indexingDurations: result.bulkCreateTimes,
+                  enrichmentDurations: result.enrichmentTimes,
+                  executionGap: remainingGap,
+                },
+                userError: result.userError,
+              });
+            } else if (
+              !(result.warningMessages.length > 0) &&
+              !(wrapperWarnings.length > 0) &&
+              !(wrapperErrors.length > 0)
```

Comment:
The wrapper errors length check here is unnecessary due to the check on line 505, right?